docs/docs/dev_docs/testing/cheat_codes.md

```diff
@@ -0,0 +1,466 @@
+
+// Computes the slot value for a given map and key on Ethereum. A convenient wrapper to find the appropriate storage slot to load or overwrite the state.
+public keccak256(baseSlot: bigint, key: bigint): bigint
+
+// Let you send transactions on Ethereum impersonating an externally owned or contract, without knowing the private key.
+public async startImpersonating(who: EthAddress): Promise<void> 
+
+// Stop impersonating an account on Ethereum that you are currently impersonating.
+public async stopImpersonating(who: EthAddress): Promise<void> 
+
+// Set the bytecode for a Ethereum contract
+public async etch(contract: EthAddress, bytecode: `0x${string}`): Promise<void> 
+
+// Get the bytecode for a Ethereum contract
+public async getBytecode(contract: EthAddress): Promise<`0x${string}`> 
+```
+
+### blockNumber
+
+#### Function Signature 
+```ts
+public async blockNumber(): Promise<number>
+```
+
+#### Description
+Fetches the current Ethereum block number.
+
+#### Example
+```ts
+const blockNumber = await cc.eth.blockNumber()
+```
+
+### chainId
+
+#### Function Signature 
+```ts
+public async chainId(): Promise<number>  
+```
+
+#### Description
+Fetches the Ethereum chain ID
+
+#### Example 
+```ts
+const chainId = await cc.eth.chainId()
+```
+
+### timestamp
+
+#### Function Signature
+```ts
+public async timestamp(): Promise<number> 
+```
+
+#### Description 
+Fetches the current Ethereum timestamp.
+
+#### Example
+```ts 
+const timestamp = await cc.eth.timestamp()
+```
+
+### mine
+
+#### Function Signature
+```ts
+public async mine(numberOfBlocks = 1): Promise<void> 
+```
+
+#### Description
+Mines the specified number of blocks on Ethereum (default 1).
+
+#### Example
+```ts
+const blockNum = await cc.eth.blockNumber();
+await cc.eth.mine(10) // mines 10 blocks
+const newBlockNum = await cc.eth.blockNumber(); // = blockNum + 10.
+```
+
+### setNextBlockTimestamp
+
+#### Function Signature  
+```ts
+public async setNextBlockTimestamp(timestamp: number): Promise<void>
+```
+
+#### Description
+Sets the timestamp (unix format in seconds) for the next mined block on Ethereum. 
+Time can only be set in the future.
+If you set the timestamp to a time in the past, this method will throw an error.
+
+#### Example
+```ts
+// // Set next block timestamp to 16 Aug 2023 10:54:30 GMT
+await cc.eth.setNextBlockTimestamp(1692183270) 
+// next transaction you will do will have the timestamp as 1692183270
+```
+
+### dumpChainState
+
+#### Function Signature
+```ts
+public async dumpChainState(fileName: string): Promise<void> 
+```
+
+#### Description
+Dumps the current Ethereum chain state to a file.
+Stores a hex string representing the complete state of the chain in a file with the provided path. Can be re-imported into a fresh/restarted instance of Anvil to reattain the same state.
+When combined with `loadChainState()` cheatcode, it can be let you easily import the current state of mainnet into the Anvil instance of the sandbox.
```

Comment:
The wording is a little weird here, `it can be let you` and it might not be the `current` state. Maybe fine to point to the other cheatcode and say use in combination.

```suggestion
To be used in combination with `loadChainState()`
```